v22.1/row-level-ttl.md

```diff
@@ -0,0 +1,342 @@
+
+- Outbox pattern: When events are written to an outbox table and published to an external system like [Kafka](https://en.wikipedia.org/wiki/Apache_Kafka), those events must be deleted to prevent unbounded growth in the size of the outbox table.
+
+## How it works
+
+At a high level, Row-Level TTL works by:
+
+- Issuing a [selection query](selection-queries.html) at a [historical timestamp](as-of-system-time.html), yielding a set of rows that are eligible for deletion.
+- Given the set of rows eligible for deletion, split what would logically be a single [`DELETE`](delete.html) statement to remove those rows into multiple SQL queries that will eventually have the equivalent effect by operating on a subset of the eligible rows.
+- As part of the above process, deciding how many rows to [`SELECT`](select-clause.html) and [`DELETE`](delete.html) at once in each of the above queries.
+- Running the SQL queries described above in parallel as [background jobs](show-jobs.html).
+- To minimize the performance impact on foreground application queries, the background deletion queries are rate limited; they are also submitted at a lower priority level using the [admission control system](architecture/admission-control.html).
+
+The process above is conceptually similar to the process described by [Batch delete on an indexed column](bulk-delete-data.html#batch-delete-on-an-indexed-column), except that Row-Level TTL is built into CockroachDB, so it saves you from having to write code to manage the process from your application and/or external job processing framework, including tuning the rate and performance of your background queries so they don't affect foreground application query performance.
+
+## When are rows deleted?
+
+Once rows are expired (that is, are older than the specified [TTL interval](#param-ttl-expire-after)), they are eligible to be deleted. However, eligible rows may not be deleted right away - instead, they are scheduled for deletion using a [background job](#view-ttl-jobs) that is run at the interval defined by the `ttl_job_cron` [storage parameter](#ttl-storage-parameters).
+
+## Syntax overview
+
+TTLs are defined on a per-table basis using SQL statements. The syntax for creating a table with an automatically managed TTL extends the [`storage_parameter` syntax](sql-grammar.html#opt_with_storage_parameter_list). For example, the SQL statement
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE ttl_test (
+  id UUID PRIMARY KEY default gen_random_uuid(),
+  description TEXT,
+  inserted_at TIMESTAMP default current_timestamp()
+) WITH (ttl_expire_after = '3 minutes');
+~~~
+
+has the following effects:
+
+<a name="crdb-internal-expiration"></a>
+
+1. Creates a repeating [scheduled job](#view-ttl-jobs) for the `ttl_test` table.
+2. Adds a hidden column `crdb_internal_expiration` of type [`TIMESTAMPTZ`](timestamp.html) to represent the TTL.
```

Comment:
do we have a hyperlink to what a hidden column does (i.e. doesn't show up on `*`). if so, let's link to that!

Reply:
I looked and we actually don't appear to!  Filed https://cockroachlabs.atlassian.net/browse/DOC-3200 to address as a followup task (not sure when, but it's filed at least :-} )

Reply:
following some PR advice, i think we should follow the DDL here and use `NOT VISIBLE` column externally here